File: source/resources/examples/typescript/transfer/SendingATransferTransactionWithMultipletokens.js
'use strict';
/*
 *
 * Copyright 2018-present BITXOR
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */
Object.defineProperty(exports, '__esModule', { value: true });
const bitxor_sdk_1 = require('bitxor-sdk');
// Retrieve from node's /network/properties or RepositoryFactory
const epochAdjustment = 123456789;
// replace with recipient address
const rawAddress = 'BXRQ5E-YACWBP-CXKGIL-I6XWCH-DRFLTB-KUK34I-YJQ';
const recipientAddress = bitxor_sdk_1.Address.createFromRawAddress(rawAddress);
// replace with network type
const networkType = bitxor_sdk_1.NetworkType.TEST_NET;
const transferTransaction = bitxor_sdk_1.TransferTransaction.create(
    bitxor_sdk_1.Deadline.create(epochAdjustment),
    recipientAddress,
    /* start block 01 */
    [
        new bitxor_sdk_1.Token(
            new bitxor_sdk_1.TokenId('7CDF3B117A3C40CC'),
            bitxor_sdk_1.UInt64.fromUint(1000),
        ),
        new bitxor_sdk_1.Token(
            new bitxor_sdk_1.TokenId('5E62990DCAC5BE8A'),
            bitxor_sdk_1.UInt64.fromUint(10 * Math.pow(10, 6)),
        ),
    ],
    /* end block 01 */
    bitxor_sdk_1.PlainMessage.create('This is a test message'),
    networkType,
    bitxor_sdk_1.UInt64.fromUint(2000000),
);
console.log(transferTransaction);

File: source/resources/examples/java/src/test/java/bitxor/guides/examples/transfer/SendingATransferTransactionWithMultipletokens.java
/*
 *
 * Copyright 2018-present BITXOR
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */

package bitxor.guides.examples.transfer;

import io.bitxor.sdk.api.RepositoryFactory;
import io.bitxor.sdk.infrastructure.vertx.RepositoryFactoryVertxImpl;
import io.bitxor.sdk.model.account.Address;
import io.bitxor.sdk.model.account.UnresolvedAddress;
import io.bitxor.sdk.model.network.NetworkType;
import io.bitxor.sdk.model.message.PlainMessage;
import io.bitxor.sdk.model.token.Token;
import io.bitxor.sdk.model.token.TokenId;
import io.bitxor.sdk.model.transaction.TransferTransactionFactory;
import org.junit.jupiter.api.Test;

import java.math.BigInteger;
import java.util.Arrays;
import java.util.concurrent.ExecutionException;

class SendingATransferTransactionWithMultipleTokens {

    @Test
    void sendingATransferTransactionWithMultipleTokens()
            throws ExecutionException, InterruptedException {
        // replace with node endpoint
        try (final RepositoryFactory repositoryFactory = new RepositoryFactoryVertxImpl(
                "NODE_URL")) {
            // replace with recipient address
            final String rawAddress = "BXRQ5E-YACWBP-CXKGIL-I6XWCH-DRFLTB-KUK34I-YJQ";
            final UnresolvedAddress recipientAddress = Address.createFromRawAddress(rawAddress);
            final NetworkType networkType = repositoryFactory.getNetworkType().toFuture().get();

            TransferTransactionFactory
                    .create(
                            networkType,
                            recipientAddress,
                            /* start block 01 */
                            Arrays.asList(new Token(new TokenId("7CDF3B117A3C40CC"),
                                            BigInteger.valueOf(1000)),
                                    new Token(new TokenId("5E62990DCAC5BE8A"),
                                            BigInteger.valueOf(10000000))),
                            /* end block 01 */
                            PlainMessage.create("This is a test message"))
                    .maxFee(BigInteger.valueOf(2000000)).build();
        }
    }
}

